### PR TITLE
Adapted to exception for vlan usage

### LINKED ISSUE #98: Adpat vlan usage to exception raising from Interface
Interface `use_tags()` and `make_tags_available()` are going to raise errors instead of returning them.

### PR DESCRIPTION
**Pull request by Alopalao**
Closes #98 

### Summary

Adapted main methods that are Interfaces related to capture `KytosTagsAreNotAvailable` and `KytosTagsNotInTagRanges`

### Local Tests

Updated unit tests
Raised exceptions manually

### End-To-End Tests
```
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.3.0
rootdir: /tests
plugins: timeout-2.1.0, rerunfailures-10.2, anyio-3.6.2
collected 244 items

tests/test_e2e_01_kytos_startup.py ..                                    [  0%]
tests/test_e2e_05_topology.py ..................                         [  8%]
tests/test_e2e_10_mef_eline.py ..........ss.....x.....x................  [ 24%]
tests/test_e2e_11_mef_eline.py ......                                    [ 27%]
tests/test_e2e_12_mef_eline.py .....Xx.                                  [ 30%]
tests/test_e2e_13_mef_eline.py ....Xs.s.....Xs.s.XXxX.xxxx..X........... [ 47%]
.                                                                        [ 47%]
tests/test_e2e_14_mef_eline.py x                                         [ 47%]
tests/test_e2e_15_mef_eline.py ....                                      [ 49%]
tests/test_e2e_20_flow_manager.py .....................                  [ 58%]
tests/test_e2e_21_flow_manager.py ...                                    [ 59%]
tests/test_e2e_22_flow_manager.py ...............                        [ 65%]
tests/test_e2e_23_flow_manager.py ..............                         [ 71%]
tests/test_e2e_30_of_lldp.py ....                                        [ 72%]
tests/test_e2e_31_of_lldp.py ...                                         [ 74%]
tests/test_e2e_32_of_lldp.py ...                                         [ 75%]
tests/test_e2e_40_sdntrace.py .............                              [ 80%]
tests/test_e2e_41_kytos_auth.py ........                                 [ 84%]
tests/test_e2e_42_sdntrace.py ..                                         [ 84%]
tests/test_e2e_50_maintenance.py ........................                [ 94%]
tests/test_e2e_60_of_multi_table.py .....                                [ 96%]
tests/test_e2e_70_kytos_stats.py ........                                [100%]
```
Note:
This PR relies on [kytos](https://github.com/kytos-ng/kytos/pull/428) PR.

### CODE REVIEW

File: requirements/dev.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
 -e git+https://github.com/kytos-ng/python-openflow.git#egg=python-openflow
--e git+https://github.com/kytos-ng/kytos.git#egg=kytos[dev]
+-e git+https://github.com/kytos-ng/kytos.git@epic/vlan_range#egg=kytos[dev]
```
Comment (Alopalao) [inline requirements/dev.in:2]: Reminder to delete this before merging.

**Review by viniarck — COMMENTED** [review]
LGTM. It's great to have the new exceptions, and also still have the conflict data when it ever happens. I'll approve this PR once the PR it depends on is ready to land. No need to update the changelog in this case since it's mostly refactoring.

**Review by viniarck — COMMENTED** [review]
It's really nice how `KytosTagError` and overall exceptions have been structured. I'll approve this PR once the kytos PR it depends on is also approved.

**Review by viniarck — APPROVED** [review]
I'll leave this pre-approved, let's revert https://github.com/kytos-ng/of_lldp/pull/99#discussion_r1376881004 as Aldo mentioned when the PR this PR depends on https://github.com/kytos-ng/kytos/pull/428. Once Kytos PR 428 is merged, let's merge this one right after.

**Pull request merged**